dart/lib/src/environment_variables.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:meta/meta.dart';
+
+import 'sentry_options.dart';
+
+/// This method reads available environment variables and uses them
+/// accordingly.
+/// To see which environment variables are available, see [EnvironmentVariables]
+///
+/// The precendence of these options are tricky,
+/// see https://docs.sentry.io/platforms/dart/configuration/options/
+/// and https://github.com/getsentry/sentry-dart/issues/306
+@internal
+void setEnvironmentVariables(SentryOptions options, EnvironmentVariables vars) {
```

Comment:
what's the inspiration for being a non-instance method? could be just a private method on the Sentry static class

Reply:
That way I can test it more easily. If it is a private method on `Sentry` I can't easily pass `MockEnvironmentVariables` to it. Well I could by adding it as parameter to `Sentry.init` but that doesn't seem that good.

What would you suggest instead?

Reply:
Another solution is that `EnvironmentVariables` could be property on SentryOptions just like `PlatformChecker`

Reply:
> Another solution is that EnvironmentVariables could be property on SentryOptions just like PlatformChecker

I think this one makes more sense